rust/cargo/BUILD

```diff
@@ -0,0 +1,3 @@
+exports_files([
+    "Cargo.toml",
```

Comment:
This Cargo.toml file is part of our under-development work where we intend to build a crates.io release from bazel; I don't think we expected it to be used for the Bazel flow when it was introduced.

Looking at the Cargo.toml though the only maybe-problematic thing is that it declares a version number for the `protobuf` crate and its going to ~always be arbitrary and wrong on main. But I suppose that is effectively true anyway for all Cargo.toml on main branches?

Reply:
Does that mean it's only supposed to be used by this module? if end users are not supposed to use this stuff can it be marked `dev_dependency = True` in MODULE.bazel?

Reply:
You

> This Cargo.toml file is part of our under-development work where we intend to build a crates.io release from bazel; I don't think we expected it to be used for the Bazel flow when it was introduced.
> 
> Looking at the Cargo.toml though the only maybe-problematic thing is that it declares a version number for the `protobuf` crate and its going to ~always be arbitrary and wrong on main. But I suppose that is effectively true anyway for all Cargo.toml on main branches?

Correct, I think the right way is to create a workspace Cargo.toml and delcare dependencies thhere and have the sub crates depends on the dependencies from the workspace. I'll amend.

Reply:
If this is only used to build the crates.io as part of our release process, marking as a dev dependency seems appropriate.

The version being a tentative -dev version at head is always the case, but we can add that file to our release process's version updaters so that the version number is updated to the proper version on release (and stays at a sensible "-dev" version).